ddtrace/contrib/openai/patch.py

```diff
@@ -0,0 +1,205 @@
+from ddtrace import config
+from ddtrace.contrib.trace_utils import set_flattened_tags
+from ddtrace.internal.agent import get_stats_url
+from ddtrace.internal.constants import COMPONENT
+from ddtrace.internal.dogstatsd import get_dogstatsd_client
+
+from .. import trace_utils
+from .. import trace_utils_async
+from ...pin import Pin
+from ..trace_utils import wrap
+from ._utils import append_tag_prefixes
+from ._utils import engine_resource_name
+from ._utils import process_request
+from ._utils import process_response
+from ._utils import process_text
+
+
+config._add(
+    "openai",
+    {
+        "_default_service": "openai",
+    },
+)
+
+REQUEST_TAG_PREFIX = "request"
+RESPONSE_TAG_PREFIX = "response"
+ERROR_TAG_PREFIX = "error"
+ENGINE = "engine"
+
+_statsd = None
+
+
+def _stats_client():
+    global _statsd
+    if _statsd is None:
+        # FIXME: this currently does not consider if the tracer
+        # is configured to use a different hostname.
+        # eg. tracer.configure(host="new-hostname")
+        _statsd = get_dogstatsd_client(
+            get_stats_url(),
+            namespace="openai",
+            tags=["env:%s" % config.env, "service:%s" % config.service, "version:%s" % config.version],
+        )
+    return _statsd
+
+
+def patch():
+    # Avoid importing openai at the module level, eventually will be an import hook
```

Comment:
Doesn't our patch-on-import monkey patching strategy mean openai will be already loaded at this time?

Reply:
yes, but inlining the import here ensures the integration is structured to work without the import (proper patch on import)